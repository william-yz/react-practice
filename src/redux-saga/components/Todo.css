.editable-cell {
  position: relative;
  user-select: none;
}

.editable-cell-input-wrapper,
.editable-cell-text-wrapper {
  padding-right: 24px;
}

.editable-cell-text-wrapper {
  padding: 5px 24px 5px 5px;
}
.delete-cell-icon {
  position: absolute;
  right: 20px;
  width: 20px;
  cursor: pointer;
}
.editable-cell-icon,
.editable-cell-icon-check {
  position: absolute;
  right: 0;
  width: 20px;
  cursor: pointer;
}

.delete-cell-icon,
.editable-cell-icon {
  line-height: 18px;
  display: none;
}


.editable-cell-icon-check {
  line-height: 28px;
}

.editable-cell:hover .editable-cell-icon {
  display: inline-block;
}

.editable-cell:hover .delete-cell-icon {
  display: inline-block;
}

.delete-cell-icon:hover,
.editable-cell-icon:hover,
.editable-cell-icon-check:hover {
  color: #108ee9;
}
