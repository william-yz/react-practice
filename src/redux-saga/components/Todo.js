import React, { PropTypes } from 'react'

import { Input, Icon } from 'antd'
import './Todo.css'

export default function Todo({
  save,
  edit,
  input,
  removeTodo,
  editting,
  content,
  complated,
  otherIsEditting }) {
  return (
    <div className="editable-cell">
      {
          editting ?
            <div className="editable-cell-input-wrapper">
              <Input
                value={content}
                onChange={input}
              />
              <Icon
                type="check"
                className="editable-cell-icon-check"
                onClick={save}
              />
            </div>
          :
          <div className="editable-cell-input-wrapper" style={{ textDecoration: complated ? 'line-through' : 'none' }}>
            {content}
            {
              (!complated && !otherIsEditting) && (
                <span>
                  <Icon
                    type="edit"
                    className="editable-cell-icon"
                    onClick={edit}
                  />
                  <Icon
                    type="delete"
                    className="delete-cell-icon"
                    onClick={removeTodo}
                  />
                </span>
              )
            }
          </div>
        }
    </div>
  )
}

Todo.propTypes = {
  save: PropTypes.func.isRequired,
  edit: PropTypes.func.isRequired,
  input: PropTypes.func.isRequired,
  removeTodo: PropTypes.func.isRequired,
  editting: PropTypes.bool.isRequired,
  content: PropTypes.string.isRequired,
  complated: PropTypes.bool.isRequired,
  otherIsEditting: PropTypes.bool.isRequired,
}
