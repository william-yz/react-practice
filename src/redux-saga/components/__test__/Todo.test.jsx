import React from 'react'
import renderer from 'react-test-renderer'

import Todo from '../Todo'

test('Todo case 1', () => {
  const props = {
    save: jest.fn(),
    edit: jest.fn(),
    input: jest.fn(),
    removeTodo: jest.fn(),
    editting: true,
    content: 'abcd',
    complated: false,
    otherIsEditting: false,
  }
  const component = renderer.create(
    <Todo
      {...props}
    />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})

test('Todo case 2', () => {
  const props = {
    save: jest.fn(),
    edit: jest.fn(),
    input: jest.fn(),
    removeTodo: jest.fn(),
    editting: false,
    content: 'abcd',
    complated: false,
    otherIsEditting: false,
  }
  const component = renderer.create(
    <Todo
      {...props}
    />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})

test('Todo case 3', () => {
  const props = {
    save: jest.fn(),
    edit: jest.fn(),
    input: jest.fn(),
    removeTodo: jest.fn(),
    editting: false,
    content: 'abcd',
    complated: true,
    otherIsEditting: false,
  }
  const component = renderer.create(
    <Todo
      {...props}
    />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})
