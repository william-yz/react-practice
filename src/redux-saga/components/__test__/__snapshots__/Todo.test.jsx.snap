exports[`test Todo case 1 1`] = `
<div
  className="editable-cell">
  <div
    className="editable-cell-input-wrapper">
    <input
      className="ant-input"
      disabled={false}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value="abcd" />
    <i
      className="anticon anticon-check editable-cell-icon-check"
      onClick={[Function]} />
  </div>
</div>
`;

exports[`test Todo case 2 1`] = `
<div
  className="editable-cell">
  <div
    className="editable-cell-input-wrapper"
    style={
      Object {
        "textDecoration": "none",
      }
    }>
    abcd
    <span>
      <i
        className="anticon anticon-edit editable-cell-icon"
        onClick={[Function]} />
      <i
        className="anticon anticon-delete delete-cell-icon"
        onClick={[Function]} />
    </span>
  </div>
</div>
`;

exports[`test Todo case 3 1`] = `
<div
  className="editable-cell">
  <div
    className="editable-cell-input-wrapper"
    style={
      Object {
        "textDecoration": "line-through",
      }
    }>
    abcd
  </div>
</div>
`;
